hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccount.java

```diff
@@ -219,6 +219,15 @@ public void setAlias(final ByteString alias) {
 		state().setAlias(alias);
 	}
 
+	public long getTransactionCounter() {
+		return state().transactionCounter();
+	}
+
+	public void setTransactionCounter(final long transactionCounter) {
+		throwIfImmutable("Cannot change this account's transactionCounter if it's immutable.");
```

Comment:
Are we sure that we shouldn't increment the `transactionCounter` on immutable accounts?

Reply:
Updating the nonce is a mutation.  The only hedge we may want to consider is if the value being set is not a change, but I think it's cleaner to reject all attempts on an immutable copy.